src/core/store/common/tokensByAsset.selectors.ts

```diff
@@ -0,0 +1,84 @@
+import { createSelector } from '@reduxjs/toolkit';
+import { memoize } from 'lodash';
+
+import { toBN } from '@utils';
+import { getConfig } from '@config';
+import { TokenView, Address } from '@types';
+
+import { VaultsSelectors } from '../modules/vaults/vaults.selectors';
+import { LabsSelectors } from '../modules/labs/labs.selectors';
+import { TokensSelectors } from '../modules/tokens/tokens.selectors';
+import { AppSelectors } from '../modules/app/app.selectors';
+import { createToken } from '../modules/tokens/tokens.selectors';
+
+const { selectVaultsMap } = VaultsSelectors;
+const { selectLabsMap } = LabsSelectors;
+const { selectTokensMap, selectTokensUser } = TokensSelectors;
+const { selectServicesEnabled } = AppSelectors;
+
+export const selectDepositTokenOptionsByAsset = createSelector(
+  [selectVaultsMap, selectLabsMap, selectTokensMap, selectTokensUser, selectServicesEnabled],
+  (vaultsMap, labsMap, tokensMap, tokensUser, servicesEnabled) =>
+    memoize((assetAddress?: string): TokenView[] => {
+      if (!assetAddress) return [];
+
+      const { userTokensAddresses, userTokensMap, userTokensAllowancesMap } = tokensUser;
+      const assetData = vaultsMap[assetAddress] ? vaultsMap[assetAddress] : labsMap[assetAddress];
+      if (!assetData) return [];
+
+      const zapperDisabled = !servicesEnabled.zapper && assetData.metadata.zapInWith === 'zapperZapIn';
+      const depositTokenAddresses: Address[] = [];
+      let mainVaultToken: Address;
+      if (zapperDisabled) {
+        mainVaultToken = assetData.token;
+      } else {
+        mainVaultToken = assetData.metadata.defaultDisplayToken;
+        depositTokenAddresses.push(...userTokensAddresses.filter((address) => address !== mainVaultToken));
+      }
```

Comment:
Now that we've simplified the logic inside the `if/else` we can remove it all together, and also the `let` and make it a `const` for readability, something like:

```suggestion
      const mainVaultToken: Address = zapperDisabled ? assetData.token : assetData.metadata.defaultDisplayToken;

      if (!zapperDisabled) {
        const { ZAP_OUT_TOKENS } = getConfig();
        withdrawTokenAddresses.push(...ZAP_OUT_TOKENS.filter((address) => address !== mainVaultToken));
      }
```

Reply:
great suggestion. refactored here https://github.com/yearn/yearn-finance-v3/pull/674/commits/0e496c73bdac14363d64c3907f39c3108889835b

Reply:
Ah, nice idea with the refactor 👍  much better!